src/definitions/modules/nag.js

```diff
@@ -94,47 +94,47 @@ $.fn.nag = function(parameters) {
 
         show: function() {
           if( module.should.show() && !$module.is(':visible') ) {
+            if(settings.onShow.call(element) === false) {
+              module.debug('onShow callback returned false, cancelling nag animation');
+              return false;
+            }
             module.debug('Showing nag', settings.animation.show);
-            if(settings.animation.show == 'fade') {
+            if(settings.animation.show === 'fade') {
               $module
-                .fadeIn(settings.duration, settings.easing)
+                .fadeIn(settings.duration, settings.easing, settings.onVisible)
               ;
             }
             else {
               $module
-                .slideDown(settings.duration, settings.easing)
+                .slideDown(settings.duration, settings.easing, settings.onVisible)
               ;
             }
           }
         },
 
         hide: function() {
-          module.debug('Showing nag', settings.animation.hide);
-          if(settings.animation.hide == 'fade') {
+          if(settings.onHide.call(element) === false) {
+            module.debug('onHide callback returned false, cancelling nag animation');
+            return false;
+          }
+          module.debug('Hiding nag', settings.animation.hide);
+          if(settings.animation.hide === 'fade') {
             $module
-              .fadeOut(settings.duration, settings.easing)
+              .fadeOut(settings.duration, settings.easing, settings.onHidden)
             ;
           }
           else {
             $module
-              .slideUp(settings.duration, settings.easing)
+              .slideUp(settings.duration, settings.easing, settings.onHidden)
             ;
           }
         },
 
-        onHide: function() {
-          module.debug('Removing nag', settings.animation.hide);
-          $module.remove();
-          if (settings.onHide) {
-            settings.onHide();
-          }
-        },
-
         dismiss: function(event) {
-          if(settings.storageMethod) {
+          if(module.hide() !== false && settings.storageMethod) {
+            module.debug('Dismissing nag', settings.storageMethod, settings.key, settings.value, settings.expires);
             module.storage.set(settings.key, settings.value);
```

Comment:
Is it safe to skip `module.storage.gset` when `module.hide() === false` ?

Reply:
Yes, the only situation where the storage should set the dismisss value is of course when hiding the nag is allowed. (if i cant hide it, i also cannot dismiss it, most logical)